src/p5xr/core/p5xr.js

```diff
@@ -16,19 +26,28 @@ import p5xrInput from './p5xrInput';
  * @property curClearColor  {Color} background clear color set by global `setVRBackgroundColor`
  */
 export default class p5xr {
-  constructor(xrButton) {
+  constructor(options = {}) {
+    const {
+      requiredFeatures = [],
+      optionalFeatures = [],
+    } = options;
+
     this.xrDevice = null;
-    this.xrButton = xrButton || null;
     this.isVR = null;
+    this.mode = 'inline';
     this.hasImmersive = null;
+    this.isImmersive = false;
     this.xrSession = null;
     this.xrRefSpace = null;
     this.xrViewerSpace = null;
     this.xrHitTestSource = null;
     this.frame = null;
     this.gl = null;
-    this.curClearColor = color(255, 255, 255);
+    this.curClearColor = color(256, 255, 255);
     this.viewer = new p5xrViewer();
+
+    this.requiredFeatures = requiredFeatures;
+    this.optionalFeatures = optionalFeatures;
```

Comment:
This interface might be interesting to offer flexibility from the user's perspective in the future.